src/synapse_tools/firewall_update.py

```diff
@@ -0,0 +1,94 @@
+# -*- coding: utf-8 -*-
+from __future__ import absolute_import
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import argparse
+import os.path
+import time
+from collections import defaultdict
+
+from inotify.adapters import Inotify
+from inotify.constants import IN_MODIFY
+from inotify.constants import IN_MOVED_TO
+
+from synapse_tools import configure_synapse
+from paasta_tools.marathon_tools import load_marathon_service_config
+from paasta_tools.marathon_tools import marathon_services_running_here
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import load_system_paasta_config
+
+
+UPDATE_SECS = 5
+SYNAPSE_SERVICE_DIR = b'/var/run/synapse/services'
+
+
+def smartstack_dependencies_of_running_firewalled_services(soa_dir=DEFAULT_SOA_DIR):
+    dependencies_to_services = defaultdict(list)
+
+    cluster = load_system_paasta_config().get_cluster()
+    for service, instance, port in marathon_services_running_here():  # TODO: + chronos
+        config = load_marathon_service_config(service, instance, cluster, load_deployments=False, soa_dir=soa_dir)
+
+        outbound_firewall = config.get_outbound_firewall()
+        if not outbound_firewall:
+            continue
+
+        dependencies = config.get_dependencies()
+
+        smartstack_dependencies = [d['smartstack'] for d in dependencies if d.get('smartstack')]
+        for smartstack_dependency in smartstack_dependencies:
+            dependencies_to_services[smartstack_dependency].append((service, instance))
+
+    return dependencies_to_services
+
+
+def parse_args(argv):
+    parser = argparse.ArgumentParser(description='Monitor synapse changes and update service firewall rules')
+    parser.add_argument('-c', '--synapse-tools-config', dest="synapse_tools_config",
+                        default='/etc/synapse/synapse-tools.conf.json',
+                        help="Path to synapse tools config (default %(default)s)")
+    parser.add_argument('-d', '--soa-dir', dest="soa_dir", metavar="soa_dir",
```

Comment:
maybe `-y`/`--yelpsoa-config-root` to match the paasta cli? (or is there some other prior art for `-d`?)